base/float.jl

```diff
@@ -414,19 +414,19 @@ end
 
 isequal(x::T, y::T) where {T<:IEEEFloat} = fpiseq(x, y)
 
-# interpret as sign-magnitude integer
-@inline function _fpint(x)
-    IntT = inttype(typeof(x))
-    ix = reinterpret(IntT, x)
-    return ifelse(ix < zero(IntT), ix ⊻ typemax(IntT), ix)
-end
-
 @inline function isless(a::T, b::T) where T<:IEEEFloat
```

Comment:
I have a hard time believing that we can do any better than LLVM/hardware level IEEE floating-point comparison in general. Is there an LLVM intrinsic we can benchmark against?

Reply:
I don't know what LLVM intrinsics are available. But there isn't any x86 instruction for what we want. x86 comparisons and vminsd/vmaxsd have the wrong semantics with regard to signed zeros and NaNs.

This change was primarily intended to the `flipifneg` operation for re-use in the fast `min`/`max`. This was the fastest version of about 5 different methods I tried.

Reply:
Lack of hardware support is surprising, but I believe you.